install.sh

```diff
@@ -85,55 +170,104 @@ if !([ -f /etc/ros/rosdep/sources.list.d/20-default.list ]); then
-    echo "source $CATKIN_DIR/devel/setup.bash" >> ~/.bashrc
-fi
+source ../devel/setup.bash
 
-# Check if the sub is set up; if it isn't, set it up
-if !(ls "$CATKIN_DIR/src" | grep --quiet "Sub8"); then
-    instlog "Looks like you don't have the sub set up, let's do that"
-    cd "$CATKIN_DIR/src"
-    git clone -q https://github.com/uf-mil/Sub8.git
-    cd Sub8
+# Clones the Sub8 directory unless it already exists
+if !(ls $CATKIN_DIR/src | grep --quiet "Sub8"); then
+    instlog "Downloading the Sub8 repository"
+    cd $CATKIN_DIR/src
+    git clone https://github.com/uf-mil/Sub8.git
+    cd $CATKIN_DIR/src/Sub8
     git remote rename origin upstream
-    instlog "Make sure you change your git to point to your own fork! (git remote add origin your_forks_url)"
+    instlog "Make sure you change your git origin to point to your own fork! (git remote add origin your_forks_url)"
 fi
 
-# Update the hosts file if necessary
-if (env | grep --quiet "SUB=true"); then
-    $CATKIN_DIR/src/Sub8/scripts/update-hosts.bash
+# Adds Sub8 aliases to $BASHRC_FILE unless already exists
+if !(cat $BASHRC_FILE | grep --quiet "source $ALIASES_FILE"); then
+    echo "" >> $BASHRC_FILE
+    echo "# Adds Sub8 aliases to shell environment" >> $BASHRC_FILE
+    echo "source $ALIASES_FILE" >> $BASHRC_FILE
 fi
 
-# Install external dependencies with another script
-instlog "Running the get_dependencies.sh script to update external dependencies"
+#================================#
+#    Dependency Installation     #
+#================================#
+instlog "Installing Sub8 dependencies"
+
+sudo apt-get install python-scipy
+sudo apt-get install tesseract-ocr
+sudo apt-get install libusb-1.0-0-dev
+
+# Tools
+sudo apt-get install -qq sshfs
+sudo apt-get install -qq git-lfs gitk
+git lfs install --skip-smudge
+sudo apt-get install -qq tmux
+
+# Libraries needed by txros
+sudo apt-get install -qq python-twisted socat
+
+# Package management
+sudo pip install -q -U setuptools
+
+# Service identity verification
+sudo pip install -q -U service_identity
+
+# Utilities
+sudo pip install -q -U argcomplete
+sudo pip install -q -U tqdm
+sudo pip install -q -U pyasn1
+sudo pip install -q -U characteristic
+sudo pip install -q -U progressbar
+
+# Machine Learning
+sudo pip install -q -U scikit-learn > /dev/null 2>&1
+
+# Visualization
+sudo pip install -q -U mayavi > /dev/null 2>&1
+
+instlog "Cloning common GIT repositories"
 cd $CATKIN_DIR/src
-$CATKIN_DIR/src/Sub8/scripts/get_dependencies.sh
+git clone https://github.com/RustyBamboo/camera1394.git
+git clone https://github.com/txros/txros.git
 
-# Attempt to build the sub from scratch on client machines
-if !(env | grep --quiet "SEMAPHORE=true"); then
-    instlog "Building the sub's software stack with catkin_make"
-    catkin_make -C "$CATKIN_DIR" -j8
-fi
+mv $CATKIN_DIR/src/Sub8/gnc/sub8_trajectory_generator $CATKIN_DIR/src/Sub8/gnc/.sub8_trajectory_generator
+
+catkin_make -C $CATKIN_DIR -B
```

Comment:
Semaphore is gud.

Not sure what the solution is for the camera1394 depending on driver_common without building twice...